cmd/slsa-github-generator/attest.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2022 SLSA Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"regexp"
+	"strings"
+
+	intoto "github.com/in-toto/in-toto-golang/in_toto"
+	slsav02 "github.com/in-toto/in-toto-golang/in_toto/slsa_provenance/v0.2"
+	"github.com/spf13/cobra"
+
+	"github.com/slsa-framework/slsa-github-generator/github"
+	"github.com/slsa-framework/slsa-github-generator/signing/sigstore"
+	"github.com/slsa-framework/slsa-github-generator/slsa"
+)
+
+var (
+	// shaCheck verifies a hash is has only hexidecimal digits and is 64
+	// characters long.
+	shaCheck = regexp.MustCompile(`^[a-fA-F0-9]{64}$`)
+)
+
+func parseSubjects(subjectsStr string) ([]intoto.Subject, error) {
+	var parsed []intoto.Subject
+
+	subjects := strings.Split(subjectsStr, "\n")
+	for _, s := range subjects {
+		parts := strings.SplitN(s, " ", 2)
+		if len(parts) == 0 {
```

Comment:
nit: `SplitN` will never return 0 results since the separator is non-empty. Optionally, `Cut()` may be more readable?

```go
shaDigest, name, found := strings.Cut(s, " ")
name = strings.TrimSpace(name)
```

Reply:
`strings.Fields` seems to be better for our use case so I went with that.